### ISSUE TITLE
[new options] object-property-newline: ignore function argument

### ISSUE THREAD

**Issue by andreineculau** [eslint/eslint#7396]
es6' destructuring provides an easy way to have the closest thing to named function arguments.

``` js
function foo({someArg, someOtherArg}) {...}
```

I'd like to enforce object-property-newline for everything, except function arguments, so that I can still call the function in a similar fashion as I would do with positional arguments, rather than require newlines e.g.

``` js
foo({someArg, someOtherArg});
foo(someArg, someOtherArg); // previously
foo({ // with the current behaviour of "object-property-newline: error"
  someArg,
  someOtherArg
});
```

**Comment by not-an-aardvark** [comment]
Thanks for the rule suggestion. To make it easier for people reviewing the issue, would you mind filling out the [rule change proposal template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md)?

**Comment by eslintbot** [comment]
Hi @andreineculau, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by andreineculau** [comment]
ah, missed it!

---

**What rule do you want to change?**

object-property-newline

**Does this change cause the rule to produce more or fewer warnings?**

fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

update option e.g. allow the `allowMultiplePropertiesPerLine` option to take 3 values: true/false and `args`.

**Please provide some example code that this change will affect:**

```js
foo({someArg, someOtherArg});
```

**What does the rule currently do for this code?**

`object-property-newline` will complain that the properties are not one-per-line. It has an option called `allowMultiplePropertiesPerLine` but it affects not only an argument object, but any objects.

**What will the rule do after it's changed?**

If `allowMultiplePropertiesPerLine` is set to `args`, then the js excerpt above will pass with flying colors, while this will still fail:

```js
let a = {someArg, someOtherArg};
foo(a);
```

**Comment by vitorbal** [comment]
I think modifying the `allowMultiplePropertiesPerLine` to accept `true`, `false` or the string `"args"` is kind of weird. If we can think of a better schema for this, I can get behind it.

**Comment by andreineculau** [comment]
I agree with you actually. I have merely tried to keep the proposal in sync with another rule (which I cannot remember right now, ofc).

I cannot come with a shorter option name, but how about adding the `allowMultiplePropertiesPerLineInArgs` option, so that one can set this rule to

```yaml
object-property-new-line
  - error
  - allowMultiplePropertiesPerLine: false # default, mentioned for completeness
    allowMultiplePropertiesPerLineInArgs: true
```

**Comment by kaicataldo** [comment]
Closing this as discussion has stalled and it doesn't appear there is enough interest, either from the team or from the community, to continue the discussion at this time. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. Please feel free to open a new issue with a proposal to continue this discussion if this is something you would like to continue advocating for.
